zerver/tests/test_events.py

```diff
@@ -3636,6 +3636,17 @@ def test_do_deactivate_user(self) -> None:
         check_subscription_peer_remove("events[0]", events[0])
         check_realm_user_update("events[1]", events[1], "is_active")
 
+        # Send peer_remove events for archived streams.
+        do_reactivate_user(user_profile, acting_user=None)
+        stream = self.make_stream("Stream to be archived")
+        self.subscribe(user_profile, "Stream to be archived")
+        do_deactivate_stream(stream, acting_user=None)
+        with self.verify_action(num_events=2) as events:
+            do_deactivate_user(user_profile, acting_user=None)
+        self.assertIn(stream.id, events[0]["stream_ids"])
+        check_subscription_peer_remove("events[0]", events[0])
+        check_realm_user_update("events[1]", events[1], "is_active")
```

Comment:
Better test design here would I think be to just create a second, archived channel rather than doing so many activate/deactivate steps, but we can clean that up later; the test uses that pattern a lot already. 

I think we could also be testing the groups state in the same `do_deactivate_user` call as the channels.